app/src/main/java/com/example/poste/models/Folder.java

```diff
@@ -97,6 +98,25 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
         }
     }
 
+    /**
+     * Deletes a given post from this folder in the data model and backend
+     *
+     * @param post the post you want to delete
+     */
+    public void deletePost(Post post) {
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.deletePost(new DeletePost(post.getId()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                posts.remove(post);
+            }
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+            }
```

Comment:
Please also introduce logging, toast messages, etc... something to help troubleshoot when this function is called

Reply:
Added Toast messages